aws/data_source_aws_route53_zone.go

```diff
@@ -170,46 +170,58 @@ func dataSourceAwsRoute53ZoneRead(d *schema.ResourceData, meta interface{}) erro
 		d.Set("linked_service_description", hostedZoneFound.LinkedService.Description)
 	}
 
-	nameServers, err := hostedZoneNameServers(idHostedZone, conn)
+	nameServers, err := hostedZoneNameServers(conn, idHostedZone, aws.StringValue(hostedZoneFound.Name))
+
 	if err != nil {
-		return fmt.Errorf("Error finding Route 53 Hosted Zone: %v", err)
+		return fmt.Errorf("error getting Route 53 Hosted Zone (%s) name servers: %w", idHostedZone, err)
```

Comment:
```suggestion
		return fmt.Errorf("getting Route 53 Hosted Zone (%s) name servers: %w", idHostedZone, err)
```
Nit: not a fan of `Error: error getting...` that this would produce.

Reply:
Thank you for raising this. Rather than leaving a lengthy comment that could be lost about the history of this, I have created https://github.com/hashicorp/terraform-provider-aws/issues/17314 with all the information. What is important right now is that `fmt.Errorf("error creating/reading/updating/deleting Service Thing (ID): Error")` is the (finally now 😖 ) [documented](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/error-handling.md) practice for consistency that's been the goal last few years, which is showing its age in Terraform 0.12 and later. I would rather continue the existing practice for later find/replace-ability until we can decide as a team on any potential new standards.

Please feel free to check that out issue and contribute!